opentelemetry-sdk/tests/trace/test_trace.py

```diff
@@ -746,64 +753,73 @@ def test_attributes(self):
             self.assertEqual(
                 root.attributes["list-of-numerics"], (123, 314, 0)
             )
+            self.assertEqual(root.attributes["mapping-key"], {"key": "value"})
+            self.assertEqual(root.attributes["array-key"], (1, 2, 3))
+            self.assertEqual(
+                root.attributes["complex-key"],
+                (
+                    {"key1": "value1"},
+                    {"key2": 42},
+                    "key3",
+                    (1, 2, 3),
+                ),
+            )
 
         attributes = {
             "attr-key": "val",
             "attr-key2": "val2",
             "attr-in-both": "span-attr",
+            "mapping-key": {"key": "value"},
+            "array-key": [1, 2, 3],
+            "complex-key": [
+                {"key1": "value1"},
+                {"key2": 42},
+                "key3",
+                [1, 2, 3],
+            ],
         }
         with self.tracer.start_as_current_span(
             "root2", attributes=attributes
         ) as root:
-            self.assertEqual(len(root.attributes), 3)
+            self.assertEqual(len(root.attributes), 6)
             self.assertEqual(root.attributes["attr-key"], "val")
             self.assertEqual(root.attributes["attr-key2"], "val2")
             self.assertEqual(root.attributes["attr-in-both"], "span-attr")
+            self.assertEqual(root.attributes["mapping-key"], {"key": "value"})
+            self.assertEqual(root.attributes["array-key"], (1, 2, 3))
+            self.assertEqual(
+                root.attributes["complex-key"],
+                (
+                    {"key1": "value1"},
+                    {"key2": 42},
+                    "key3",
+                    (1, 2, 3),
+                ),
+            )
 
     def test_invalid_attribute_values(self):
         with self.tracer.start_as_current_span("root") as root:
             with self.assertLogs(level=WARNING):
-                root.set_attributes(
-                    {"correct-value": "foo", "non-primitive-data-type": {}}
-                )
-
-            with self.assertLogs(level=WARNING):
-                root.set_attribute("non-primitive-data-type", {})
-            with self.assertLogs(level=WARNING):
-                root.set_attribute(
-                    "list-of-mixed-data-types-numeric-first",
-                    [123, False, "string"],
-                )
-            with self.assertLogs(level=WARNING):
-                root.set_attribute(
-                    "list-of-mixed-data-types-non-numeric-first",
-                    [False, 123, "string"],
-                )
-            with self.assertLogs(level=WARNING):
-                root.set_attribute(
-                    "list-with-non-primitive-data-type", [{}, 123]
-                )
-            with self.assertLogs(level=WARNING):
-                root.set_attribute("list-with-numeric-and-bool", [1, True])
```

Comment:
Maybe keep these tests but remove the `assertLogs` context manager / i.e. they should pass?